kolibri/plugins/learn/assets/src/views/action-bar-search-box/index.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+
+  <div class="search-box-wrapper">
+    <div v-show="showSearchBox" :class="{ 'search-box-dropdown': showDropdownSearchBox }">
+      <search-box
+        ref="searchBox"
+        :icon="showDropdownSearchBox ? 'arrow_forward' : 'search'"
+        :width="showDropdownSearchBox ? 'calc(100vw - 120px)' : '150px'"
+      />
+    </div>
+
+    <ui-icon-button
+      v-show="showDropdownSearchBox"
+      icon="search"
+      type="primary"
+      color="primary"
+      @click="toggleDropdownSearchBox"
+    />
+
+    <div
+      v-show="showDropdownSearchBox && searchBoxOpen"
+      class="backdrop"
+      @click="toggleDropdownSearchBox">
+    </div>
+  </div>
+
+</template>
+
+
+<script>
+
+  import responsiveWindow from 'kolibri.coreVue.mixins.responsiveWindow';
+  import { PageNames } from '../../constants';
+  import uiIconButton from 'keen-ui/src/UiIconButton';
+  import searchBox from '../search-box';
+
+  export default {
+    $trNameSpace: 'learnSearchBox',
+    $trs: {
+      search: 'Search',
+      clear: 'Clear',
+    },
+    mixins: [responsiveWindow],
+    components: {
+      uiIconButton,
+      searchBox,
+    },
+    data() {
+      return {
+        searchQuery: this.searchTerm,
+        searchBoxOpen: false,
+      };
+    },
+    computed: {
+      showDropdownSearchBox() {
+        return this.windowSize.breakpoint <= 0;
+      },
+      showSearchBox() {
+        if (this.showDropdownSearchBox) {
+          return this.searchBoxOpen;
+        }
+        return true;
+      },
+    },
+    methods: {
+      toggleDropdownSearchBox() {
+        this.searchBoxOpen = !this.searchBoxOpen;
+        if (this.searchBoxOpen) {
+          this.$nextTick(() => {
+            this.$refs.searchBox.$refs.searchInput.focus();
```

Comment:
I think the main question here is: does `$nextTick` guarantee that it's in the DOM and visible? It might be true, but I'm not sure. I think @DXCanas's point is that responding to an event is more reliable and not subject to race conditions.

another slight smell is the nested `$ref`s:

```javascript
this.$refs.searchBox.$refs.searchInput.focus();
```

One strategy might be to move the focus logic completely inside of `searchInput` and have it listen to it's own events, or try to rely on an `autofocus` attribute?